cpp/arcticdb/processing/clause.hpp

```diff
@@ -110,6 +112,62 @@ struct PassthroughClause {
     void set_processing_config(ARCTICDB_UNUSED const ProcessingConfig& processing_config) {}
 };
 
+struct TopKClause {
+    ClauseInfo clause_info_;
+    std::vector<float_t> query_vector;
+    uint8_t k;
+
+    TopKClause() = delete;
+
+    ARCTICDB_MOVE_COPY_DEFAULT(TopKClause);
+
+    TopKClause(std::vector<float_t> query_vector, uint8_t k);
+
+    [[nodiscard]] Composite<ProcessingSegment> process(
+            std::shared_ptr<Store> store,
+            Composite<ProcessingSegment> &&p
+    ) const;
+
+    [[nodiscard]] std::optional<std::vector<Composite<ProcessingSegment>>> repartition(
+            ARCTICDB_UNUSED std::vector<Composite<ProcessingSegment>> &&) const {
+        return std::nullopt;
+    }
+
+    [[nodiscard]] const ClauseInfo& clause_info() const {
+        return clause_info_;
+    }
+
+    void set_processing_config(ARCTICDB_UNUSED const ProcessingConfig& processing_config) {}
+
+
+    [[nodiscard]] std::string to_string() const;
+};
+
+struct NamedColumnSortedByDistance {
+    const double_t distance;
+    const std::shared_ptr<Column> column;
+    const std::string_view name;
+
+    explicit NamedColumnSortedByDistance(float_t similarity,
+                                         std::shared_ptr<Column> column,
+                                         std::string_view name) :
+            distance(similarity),
+            column(column),
+            name(name) {}
+
+    NamedColumnSortedByDistance() = delete;
```

Comment:
This class can be `ARCTICDB_MOVE_COPY_DEFAULT` as well

Reply:
Seemingly not since this triggers the error in Mac OS compilation [here](https://github.com/man-group/ArcticDB/actions/runs/5893310940/job/15984573949?pr=734).